core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -2448,7 +2448,7 @@ public final void clearAllSecondaryUserStores() {
     /**
      * {@inheritDoc}
      */
-    public final String[] getAllSecondaryRoles() throws UserStoreException {
+    public String[] getAllSecondaryRoles() throws UserStoreException {
```

Comment:
Even though keeping it final fine for currently extended classes, ideally this should not be final, isn't it?